src/core/CoreActionController.cpp

```diff
@@ -294,20 +281,10 @@ void CoreActionController::initExternalControlInterfaces()
 			//STRIP_VOLUME_ABSOLUTE
 			auto pInstr = pInstrList->get( i );
 			setStripVolume( i, pInstr->get_volume(), false );
-			
-			float fPan_L = pInstr->get_pan_l();
-			float fPan_R = pInstr->get_pan_r();
 
 			//PAN_ABSOLUTE
-			float fPanValue = 0.0;
-			if (fPan_R == 1.0) {
-				fPanValue = 1.0 - (fPan_L / 2.0);
-			}
-			else {
-				fPanValue = fPan_R / 2.0;
-			}
-		
-			setStripPan( i, fPanValue, false );
+			float fValue = pInstr->getPanWithRangeFrom0To1();
```

Comment:
I see. By changing the range we might break existing scripts relying on MIDI/OSC feedback of the pan in the range of [0,1]. But I think it's not as crucial as a mismatch between the pan value the user sets in the Mixerline and the one that is written in the .h2song file. What's your opinion on this?

Reply:
Well, we need those getters and setters for the midi at least.
I am not a user of osc, but I imagined that I may have broken it. So I trust on your opinion

What about the note pan properties ruler? Maybe the way it is now is ok as it has not an associated widget value?

Reply:
Ok.
You mean to write a new function `setStripPan()` for the range [-1;1]?

Reply:
> You mean to write a new function setStripPan() for the range [-1;1]?

Yes. Having a dedicated function associated with the new symmetric PAN endpoints will make things more consistently than adding an argument in `setStripPan`.

Reply:
Ok. Not to demand the job, but some info/example would speed up the process since I don't know how the osc syntax works.
I see that `OscServer` depends on `MidiAction` functions.
Is Osc some kind of messages that the user writes on the shell? If so, can values be fractional? I see they are generally assumed as `int` in [0;127]

Reply:
Okay, so first of all in order to easily test OSC commands I do recommend the `oscsend` shell program. It's bundled in the **liblo-tools** repo (AFAIR you are using a Debian-based Linux).
The following command will set the pan value of the first mixer strip to 0.9 (note that you have to enable OSC support in the PreferencesDialog (and maybe restart) first).

```bash
oscsend localhost 9000 /Hydrogen/PAN_ABSOLUTE/1 f 0.9
```

> I see that OscServer depends on MidiAction functions.
Is Osc some kind of messages that the user writes on the shell? If so, can values be fractional? I see they are generally assumed as int in [0;127]

You can think of OSC as something like an extension of MIDI to allow for various other message types, like floats, strings, and arbitrary blobs of data, to easily achieve control via a network and some other stuff. 

Unfortunately, the OSC interface is not as clean as it could be. All numerical input has to be provided as floats, even if Hydrogen just wants integer, and also functions requiring no input value at all, like stopping the playback, are registered as requiring a float input too. This is to mitigate limitations of certain OSC implementations, like TouchOSC.

`PAN_ABSOLUTE/<STRIP_NUMBER>` requires a value between 0.0 and 1.0. `PAN_RELATIVE/<STRIP_NUMBER>` either 1 or -1 (or any value other than 1). In the new version of the documentation I noted the input ranges, types, and basic information about all OSC commands. But it's not available online yet but still dormant in the other repo,

Reply:
Thanks. I have pushed a commit for this...
What would be the benefit of `PAN_RELATIVE_SYM`?

Reply:
> What would be the benefit of PAN_RELATIVE_SYM?

You're right. I guess none.  Sorry for the noise :)